x-pack/plugins/infra/public/components/log_stream/log_stream_embeddable.tsx

```diff
@@ -38,6 +39,8 @@ export class LogStreamEmbeddable extends Embeddable<LogStreamEmbeddableInput> {
     parent?: IContainer
   ) {
     super(initialInput, {}, parent);
+
+    this.getInput$().subscribe(() => this.renderComponent());
```

Comment:
This in combination with the `reload()` method lead to double rendering when "refresh" is clicked in the dashboard date picker.

It also doesn't unsubscribe when the embeddable is destroyed, so it might leak memory.

I wonder if we shouldn't stick with the architecture shown in the embeddable example plugins: `render()` mounts a component that receives as props `input$`, `output$` and a `reload$` subject, that emits the current timestamp whenever this embeddable's `reload()` is called. Then that component could subscribe to the observables (e.g. using `useObservableState`), which ensures that the subscription is coupled to the lifecycle.

Alternatively we could defer that to later and work around these problems by keeping track of the subscription manually, since the double rendering doesn't seem to cause much disruption. What do you think?

Reply:
Good catch, I didn't consider that.

For now I'm making sure there are no double reloads and that the subscription gets properly handled.

I'm not sure what the advantage would be of using a wrapping component versus what we have now (for now at least). I think we can merge this now to get it in 7.12 and explore later what the best approach is. Makes sense?